service/matching/workers/registry_test.go

```diff
@@ -195,6 +195,112 @@ func TestRegistryImpl_ListWorkers(t *testing.T) {
 	}
 }
 
+// Exercises the query matching functionality of ListWorkers.
+func TestRegistryImpl_ListWorkersWithQuery(t *testing.T) {
+	tests := []struct {
+		name            string
+		setup           func(*registryImpl)
+		nsID            namespace.ID
+		query           string
+		expectedCount   int
+		expectedWorkers []string // WorkerInstanceKeys
+		expectError     bool
+	}{
+		{
+			name: "valid query - basic filtering",
+			setup: func(r *registryImpl) {
+				r.upsertHeartbeats("namespace1", []*workerpb.WorkerHeartbeat{
+					{WorkerInstanceKey: "worker1", TaskQueue: "queue1"},
+					{WorkerInstanceKey: "worker2", TaskQueue: "queue2"},
```

Comment:
nit: can/should `TaskQueue` be dropped here if it's not relevant to the test?

Reply:
I added a test with compound query (multiple conditions) now :)